packages/components/src/presentation-components/instance-filter-builder/Types.ts

```diff
@@ -59,3 +59,20 @@ export interface PresentationInstanceFilterInfo {
   /** Classes of the properties used in filter. */
   usedClasses: ClassInfo[];
 }
+
+/**
+ * Data structure that contains information about property used for building filter.
+ * @beta
+ */
+export interface InstanceFilterPropertyInfo {
+  /** Content descriptor field that represents this property. */
+  field: PropertiesField;
+  /** Property description  */
+  propertyDescription: PropertyDescription;
+  /** Id of the class where this property is defined. */
+  sourceClassId: ClassId;
+  /** Name of the class that was used to access this property. */
+  className: string;
+  /** Label of related property category. */
+  categoryLabel?: string;
```

Comment:
Yes it could be retrieved from `field` but for some pieces we need to traverse fields hierarchy or use data from parent field if there is one. So I wanted to simplify this and create this information while creating `propertyDescription`.